Hudl.FFprobe/MediaLoader.cs

```diff
@@ -16,23 +16,53 @@ public MediaLoader(IContainer resource)
 
         public void ReadInfo(IContainer resource)
         {
+            ReadInfo(resource, LoaderFlags.ShowFormat | LoaderFlags.ShowStreams); 
+        }
+
+
+        public void ReadInfo(IContainer resource, LoaderFlags flags)
+        {
             var ffprobeCommand = FFprobeCommand.Create(resource)
-                                               .AddSetting(new ShowFormat())
-                                               .AddSetting(new ShowStreams())
-                                               .AddSetting(new PrintFormat(PrintFormat.JsonFormat))
-                                               .Execute();
+                .AddSetting(new PrintFormat(PrintFormat.JsonFormat));
+
+            if (flags.HasFlag(LoaderFlags.ShowFormat))
```

Comment:
It seems like this kind of logic would belong in the `Create` factory method, no?